### DISCUSSION TITLE
Should use getStaticProps or getServerSideProps

### DISCUSSION THREAD

**Discussion by LeoMrel** [vercel/next.js#29889] [Help] [answered]
I'm currently building a sort of clone of TMDB and I want to show the results from one of the first ten random "trendings" pages. This is my code so far, but since Nextjs makes requests everytime during development I'm not entirely sure how my code will behave once in production.

```
 export async function getStaticProps() {
  const random = Math.floor(Math.random() * 20)
  const urls = [
    fetch(`${process.env.TRENDING_URL}&page=${random}`),
    fetch(process.env.HERO_IMAGE_URL)
  ]
  const res = await Promise.all(urls);
  const results = res.map(async (result) => await result.json())
  const [data, hero] = await Promise.all(results);

  const image = hero.results[random].backdrop_path
  return {
      props: {
          data,
          image
      }
  }
}
```

**Comment by miguderp** [comment, marked as answer]
The `Math.random()` will only execute during build time and will therefore become a static float that will be served everytime.

That being said, `getStaticProps` offers a [`revalidate` option](https://nextjs.org/docs/basic-features/data-fetching#incremental-static-regeneration) so you could specify it and thanks to that every time your page will revalidate (i.e. rebuild), your `Math.random()` will return something different.

However you need to accept that, if you set your revalidate to something like 60, the `Math.random()` will be the same for whoever visits the page during those 60s (or if the same user reloads the page multiple times during those 60s).

If your server’s performance allows it, you can also have a short revalidate of 1s for example.
